File: BOJ/Base/Ex2292.java
/**
 *  2021-02-02
 */
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Ex2292 {

	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	
		int N = Integer.parseInt(br.readLine());
		
		int range = 2;
		int output= 1;
		
		
		
		if(N ==1) {
			System.out.println("1");
		}else{
			while(range <= N) {
				range = range + (6*output);
				output++;
			}
			System.out.println(output);
		}
	}

}

File: BOJ/Base/Ex2292.kt
/** 21.07.19 벌집 **/

import java.io.BufferedReader
import java.io.InputStreamReader

fun main(){
    val br = BufferedReader(InputStreamReader(System.`in`))
    val N = br.readLine().toInt()
    var i = 1
    var count = 1

    if(N == 1) {
        print((1))
        return
    }
    while( i < N){
        i += (6 * count)
        count++
    }

    print(count)

}
